### PR TITLE
Improve standard project development build scripts

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

This PR introduces some improvements on the development build scripts of standard project:

- support for configurations in package.json
```js
// eg custom themes configuration
"mapstore": { "themes": [ "default" ]  }
```
- support of mapstore as npm module/tarball

- use of html template instead of static html in project

- custom dev plugins argument in build config

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Other... Please describe: Improve project scripts

## Issue

**What is the current behavior?**

#<issue>

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by offtherailz** [comment]
This looks mergiable, even if the effective usage of this feature need some changes in the standard project build script. In fact it adds some files that aren't used yet. 
Can we list what we should add to make it work, so we can do it next week?

File: project/standard/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    base: path.join(__dirname, "..", "..", "..", ".."),
+    dist: path.join(__dirname, "..", "..", "..", ".."),
+    framework: path.join(__dirname, "..", "..", "..", "..", "node_modules", "mapstore", "web", "client"),
+    code: [
+        path.join(__dirname, "..", "..", "..", "..", "js"),
+        path.join(__dirname, "..", "..", "..", "..", "node_modules", "mapstore", "web", "client")
+    ]
+};
+
+const mapstoreConfig = require(path.join(__dirname, "..", "..", "..", "..", "package.json")).mapstore || {};
```
Comment (MV88) [inline project/standard/webpack.config.js:16]: can we document this new possibility somewhere?

**Review by MV88 — CHANGES_REQUESTED** [review]
i think is better to start documenting this changes / new features like build config in package.json of projects @allyoucanmap what do you think?

**Review by MV88 — APPROVED** [review]

File: project/standard/webpack.config.js (hunk 1 above)
Comment (allyoucanmap) [reply project/standard/webpack.config.js:51]: @MV88 I updated the PR with a comment

**Comment by allyoucanmap** [comment]
> This looks mergiable, even if the effective usage of this feature need some changes in the standard project build script. In fact it adds some files that aren't used yet.
> Can we list what we should add to make it work, so we can do it next week?
> 

@MV88 @offtherailz @mbarto 

I have no hurry to merge this PR

**Review by MV88 — APPROVED** [review]
super!

**Comment by allyoucanmap** [comment]
closed in favor of https://github.com/geosolutions-it/MapStore2/pull/6371

**Pull request closed without merging**